### PR TITLE
link unfurling Share to teams csharp [Dont Merge]

### PR DESCRIPTION
**Pull request by NikitaJain-MSFT**

### CODE REVIEW

File: samples/link-unfurling-stt/csharp/README.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+languages:
+- csharp
+extensions:
+contentType: samples
+createdDate: "14-03-2022 00:15:13"
+---
+
+# Link unfruling Share to teams sample
+
+This sample demos the feature of link unfurling in share to teams.
```
Comment (v-chetsh) [inline samples/link-unfurling-stt/csharp/README.md:17]: Explain a bot more on 'share to teams'

File: samples/link-unfurling-stt/csharp/README.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+## Key features
+  
+  ![tab](LinkUnfurlingStt/Images/Tab.PNG)
+
```
Comment (v-chetsh) [inline samples/link-unfurling-stt/csharp/README.md:22]: Update these images with proper images.
Confirm once.

File: README.md [hunk 3]
```diff
@@ -178,6 +178,7 @@ The [Teams Toolkit](https://marketplace.visualstudio.com/items?itemName=TeamsDev
 |22| Activity feed broadcast |Demonstrating a feature to notify any message to all members of the organisation using activity feed notification .|[View][graph-activity-feed-broadcast#csharp]|[View][graph-activity-feed-broadcast#js]|
```
Comment (v-chetsh) [inline README.md:178]: I can not see the deployment file in this project, Can you please add it.
Also, deployed the code if not done.

**Review by v-chetsh — COMMENTED** [review]
Reviewed and provided my comments.

File: samples/link-unfurling-stt/csharp/LinkUnfurlingStt/AppManifest/manifest.json [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+      ]
+    }
+  ],
+  "composeExtensions": [
+    {
+      "botId": "<<Microsoft-App-Id>>",
+      "canUpdateConfiguration": true,
+      "commands": [
+        {
+          "id": "searchQuery",
```
[the remark this answers is not in the document]
Comment (NikitaJain-MSFT) [reply samples/link-unfurling-stt/csharp/LinkUnfurlingStt/AppManifest/manifest.json:49]: Link unfurling is part of messaging extension feature therefore we require commands

**Review by v-chetsh — APPROVED** [review]
Changes looks fine, approving it.

File: README.md [hunk 5]
```diff
@@ -32,6 +32,8 @@ Here are all the samples which are deployed on Microsoft Azure and you can try i
 |19|Incoming Webhook|This sample is used to send card using incoming webhook.|[View](/demo%20manifest/Incoming-Webhook.zip) |
 |20|People picker control in Adaptive Cards | This sample shows the feature of people picker control in Adaptive Cards.|[View](/demo%20manifest/People-Picker-Adaptive-Card.zip) |
 |21|App complete auth|This sample demos authentication feature in bot,tab and messaging extension.|[View](/demo%20manifest/App-Complete-Auth.zip) |
+|22|Link unfurling for 'share to teams'|This sample demos the feature of link unfurling for Share to teams.|[View](/demo%20manifest/Link-Unfurling-Stt.zip) |
```
Comment (Wajeed-msft) [inline README.md:35]: Let's rename the zip file as **Link-Unfurling-In-Share-To-Teams.zip**.

File: README.md [hunk 6]
```diff
@@ -320,6 +323,7 @@ The [Teams Toolkit](https://marketplace.visualstudio.com/items?itemName=TeamsDev
 [graph-activity-feed-broadcast#js]:samples/graph-activity-feed-broadcast/nodejs
 [app-identity-link-with-sso#nodejs]:samples/app-identity-link-with-sso/nodejs
 [tab-staggered-permission#nodejs]:samples/tab-staggered-permission/nodejs
+[link-unfurling-stt#cs]:samples/link-unfurling-stt/csharp
```
Comment (Wajeed-msft) [inline README.md:326]: Could you please rename folder to **app-link-unfurling-in-share-to-stage**?

File: samples/link-unfurling-stt/csharp/LinkUnfurlingStt/AppManifest/manifest.json [hunk 7]
```diff
@@ -0,0 +1,82 @@
+﻿{
+  "$schema": "https://developer.microsoft.com/en-us/json-schemas/teams/v1.11/MicrosoftTeams.schema.json",
```
Comment (Wajeed-msft) [inline samples/link-unfurling-stt/csharp/LinkUnfurlingStt/AppManifest/manifest.json:2]: Could you please use the latest manifest version?

File: samples/link-unfurling-stt/csharp/LinkUnfurlingStt/AppManifest/manifest.json [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+          ],
+          "parameters": [
+            {
+              "name": "searchQuery",
+              "title": "Search",
+              "description": "Your search query",
+              "inputType": "text"
+            }
+          ]
+        }
```
Comment (Wajeed-msft) [inline samples/link-unfurling-stt/csharp/LinkUnfurlingStt/AppManifest/manifest.json:59]: Can we remove the search extension?

File: samples/link-unfurling-stt/csharp/LinkUnfurlingStt/Bots/ActivityBot.cs [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+        /// <summary>
+        /// Invoked when link is unfurled.
+        /// <returns> Adaptive card for link unfurling.</returns>
+        private AdaptiveCard GetUnfurlCard()
+        {
+            var card = new AdaptiveCard(new AdaptiveSchemaVersion(1, 0));
+
+            card.Body.Add(new AdaptiveTextBlock()
+            {
+                Text = "The analytics details are",
```
Comment (Wajeed-msft) [inline samples/link-unfurling-stt/csharp/LinkUnfurlingStt/Bots/ActivityBot.cs:69]:
```suggestion
                Text = "Analytics details:",
```

File: samples/link-unfurling-stt/csharp/LinkUnfurlingStt/Pages/tab.cshtml [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+@model LinkUnfurlingStt.Pages.tabModel
+@{
+}
+
+<head>
+    <script async defer src="https://teams.microsoft.com/share/launcher.js"></script>
+    <script src="https://statics.teams.cdn.office.net/sdk/v1.10.0/js/MicrosoftTeams.min.js"
+        integrity="sha384-6oUzHUqESdbT3hNPDDZUa/OunUj5SoxuMXNek1Dwe6AmChzqc6EJhjVrJ93DY/Bv"
+        crossorigin="anonymous"></script>
+    <script>
```
Comment (Wajeed-msft) [inline samples/link-unfurling-stt/csharp/LinkUnfurlingStt/Pages/tab.cshtml:11]: Do we need this?

File: samples/link-unfurling-stt/csharp/LinkUnfurlingStt/Pages/tab.cshtml [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+            margin-left: 38rem;
+        }
+        #reportimg {
+            margin-left: 28rem;
+        }
+        #stt-btn {
+            margin-left: 40rem;
+            margin-top: 2rem;
+        }
+    </style>
```
Comment (Wajeed-msft) [inline samples/link-unfurling-stt/csharp/LinkUnfurlingStt/Pages/tab.cshtml:27]: Could you please move styles to css file under wwwroot folder?

File: samples/link-unfurling-stt/csharp/LinkUnfurlingStt/Pages/tab.cshtml [hunk 12]
```diff
@@ -0,0 +1,35 @@
...
+</head>
+<body class="theme-light">
+    <div class="surface">
+        <h3 id="tabheader"> Link unfurling in STT</h3>
```
Comment (Wajeed-msft) [inline samples/link-unfurling-stt/csharp/LinkUnfurlingStt/Pages/tab.cshtml:31]:
```suggestion
        <h3 id="tabheader"> Link unfurling in Share to Teams</h3>
```

File: samples/link-unfurling-stt/csharp/README.md [hunk 13]
```diff
@@ -0,0 +1,90 @@
...
+  determine dotnet version
+  ```bash
+  dotnet --version
+  ```
+- [Ngrok](https://ngrok.com/download) (For local environment testing) Latest (any other tunneling software can also be used)
+  
+  run ngrok locally
+  ```bash
+  ngrok http -host-header=localhost 3978
+  ```
```
Comment (Wajeed-msft) [inline samples/link-unfurling-stt/csharp/README.md:40]: This is covered in section below as well. Could you please remove this one?

File: samples/link-unfurling-stt/csharp/LinkUnfurlingStt/Pages/tab.cshtml [hunk 14]
```diff
@@ -0,0 +1,35 @@
+﻿@page
```
Comment (Wajeed-msft) [inline samples/link-unfurling-stt/csharp/LinkUnfurlingStt/Pages/tab.cshtml:1]: Could you please change the casing of file name to Pascal? Please rename the tabModel  as well.

**Review by Wajeed-msft — CHANGES_REQUESTED** [review]

File: samples/link-unfurling-stt/csharp/LinkUnfurlingStt/Pages/tab.cshtml [hunk 15]
```diff
@@ -0,0 +1,35 @@
...
+            margin-left: 40rem;
+            margin-top: 2rem;
+        }
+    </style>
+</head>
+<body class="theme-light">
+    <div class="surface">
+        <h3 id="tabheader"> Link unfurling in STT</h3>
+        <img id="reportimg" src="/report.png" width="500" height="425" />
+        <div  id="stt-btn" class="teams-share-button" data-href="https://profile.botframework.com?hello" data-button-type="medium" data-icon-px-size="160" data-preview="true"></div>
```
Comment (Wajeed-msft) [inline samples/link-unfurling-stt/csharp/LinkUnfurlingStt/Pages/tab.cshtml:33]: Can we remove the reference to Bot Framework completely from solution and use the ngrok URL (ApplicationBaseUrl from applicationSettings.json)?

**Comment by v-chetsh** [comment]
@ht4963-ms - Can you please review and approve this PR?

**Comment by v-chetsh** [comment]
@ht4963-ms - Can you please review and approve this PR?

File: samples/app-link-unfurling-in-share-to-teams/csharp/README.md [hunk 16]
```diff
@@ -0,0 +1,119 @@
+---
+page_type: sample
+description: This sample demos the feature of link unfurling for Share to teams.
```
Comment (ChetanSharma-msft) [inline samples/app-link-unfurling-in-share-to-teams/csharp/README.md:3]: This sample app shows the feature of link unfurling for Share to teams.

File: samples/app-link-unfurling-in-share-to-teams/csharp/README.md [hunk 17]
```diff
@@ -0,0 +1,119 @@
...
+- office
+- office-365
+languages:
+- csharp
+extensions:
+contentType: samples
+createdDate: "14-12-2022 00:15:13"
+---
+
+# Link unfurling in Share to teams sample
```
Comment (ChetanSharma-msft) [inline samples/app-link-unfurling-in-share-to-teams/csharp/README.md:15]: `## Link unfurling in Share to teams sample`

File: samples/app-link-unfurling-in-share-to-teams/csharp/README.md [hunk 18]
```diff
@@ -0,0 +1,119 @@
...
+description: This sample demos the feature of link unfurling for Share to teams.
+products:
+- office-teams
+- office
+- office-365
+languages:
+- csharp
+extensions:
+contentType: samples
+createdDate: "14-12-2022 00:15:13"
```
Comment (ChetanSharma-msft) [inline samples/app-link-unfurling-in-share-to-teams/csharp/README.md:12]: Metadata for sample.json is missing here.

File: samples/app-link-unfurling-in-share-to-teams/csharp/assets/sample.json [hunk 19]
```diff
@@ -0,0 +1,72 @@
+[
+  {
+    "name": "officedev-microsoft-teams-samples-app-link-unfurling-in-share-to-teams-csharp",
+    "source": "officeDev",
+    "title": "App link unfurling in share to teams",
+    "shortDescription": "Demonstrating feature of link unfurling in share to teams.",
```
Comment (ChetanSharma-msft) [inline samples/app-link-unfurling-in-share-to-teams/csharp/assets/sample.json:6]: Demonstrating the feature of link unfurling in share to teams.

File: samples/app-link-unfurling-in-share-to-teams/csharp/assets/sample.json [hunk 20]
```diff
@@ -0,0 +1,72 @@
...
+    "url": "https://github.com/OfficeDev/Microsoft-Teams-Samples/tree/main/samples/app-link-unfurling-in-share-to-teams/csharp",
+    "longDescription": [
+      "Demonstrating feature of link unfurling in share to teams."
```
Comment (ChetanSharma-msft) [inline samples/app-link-unfurling-in-share-to-teams/csharp/assets/sample.json:9]: This sample app shows the feature of link unfurling in share to teams.

File: samples/app-link-unfurling-in-share-to-teams/csharp/assets/sample.json [hunk 21]
```diff
@@ -0,0 +1,72 @@
...
+        "key": "TEAMS-FEATURES",
+        "value": "tab"
+      }
+    ],
+    "thumbnails": [
+      {
+        "type": "image",
+        "order": 100,
+        "url": "https://raw.githubusercontent.com/OfficeDev/Microsoft-Teams-Samples/main/samples/app-link-unfurling-in-share-to-teams/csharp/LinkUnfurlingInShareToTeams/Images/app-link-unfurling-stt.gif",
+        "alt": "Solution UX showing app check in location"
```
Comment (ChetanSharma-msft) [inline samples/app-link-unfurling-in-share-to-teams/csharp/assets/sample.json:43]: Wrong text:
Solution UX showing the feature of link unfurling in share to teams.

**Review by ChetanSharma-msft — CHANGES_REQUESTED** [review]
Added my review comments

**Comment by ChetanSharma-msft** [comment]
Kindly run build pipeline once changes are done.

File: README.md [hunk 22]
```diff
@@ -194,6 +194,7 @@ The [Teams Toolkit](https://marketplace.visualstudio.com/items?itemName=TeamsDev
 |1|QBot          | QBot is a solution designed for classroom teaching scenarios which allows teachers, tutors, and students to intelligently answer each other's questions within the Microsoft Teams collaboration platform. |[View][msteams-app-qbot]
 |2|Resource Hub          | Resource Hub is a solution designed for all the help you need to use Teams, all in one place. |[View][msteams-app-resource-hub]
 
+|25|Link unfurling for 'share to teams'|This sample demos the feature of link unfurling for Share to teams.|[View](/demo%20manifest/Link-Unfurling-In-Share-To-Teams.zip) |
```
[the remark this answers is not in the document]
Comment (Prithvi-MSFT) [reply README.md]: Removed it

File: samples/app-link-unfurling-in-share-to-teams/csharp/assets/sample.json (hunk 20 above)
Comment (Prithvi-MSFT) [reply samples/app-link-unfurling-in-share-to-teams/csharp/assets/sample.json]: Changed it

File: samples/app-link-unfurling-in-share-to-teams/csharp/assets/sample.json (hunk 19 above)
Comment (Prithvi-MSFT) [reply samples/app-link-unfurling-in-share-to-teams/csharp/assets/sample.json]: changed it

File: samples/app-link-unfurling-in-share-to-teams/csharp/README.md (hunk 18 above)
Comment (Prithvi-MSFT) [reply samples/app-link-unfurling-in-share-to-teams/csharp/README.md:12]: updated

File: samples/app-link-unfurling-in-share-to-teams/csharp/README.md (hunk 17 above)
Comment (Prithvi-MSFT) [reply samples/app-link-unfurling-in-share-to-teams/csharp/README.md]: Changed it

File: samples/app-link-unfurling-in-share-to-teams/csharp/README.md (hunk 16 above)
Comment (Prithvi-MSFT) [reply samples/app-link-unfurling-in-share-to-teams/csharp/README.md]: Updated

File: samples/app-link-unfurling-in-share-to-teams/csharp/LinkUnfurlingInShareToTeams/AppManifest/manifest.json [hunk 23]
```diff
@@ -0,0 +1,65 @@
+﻿{
+  "$schema": "https://developer.microsoft.com/json-schemas/teams/v1.15/MicrosoftTeams.schema.json",
+  "manifestVersion": "1.15",
+  "version": "1.0.0",
+  "id": "<<Microsoft-App-Id>>",
+  "packageName": "com.microsoft.teams.linkunfurlingstt",
+  "developer": {
+    "name": "Microsoft",
+    "websiteUrl": "https://example.azurewebsites.net",
```
[the remark this answers is not in the document]
Comment (Prithvi-MSFT) [reply samples/app-link-unfurling-in-share-to-teams/csharp/LinkUnfurlingInShareToTeams/AppManifest/manifest.json]: changed

File: samples/app-link-unfurling-in-share-to-teams/csharp/assets/sample.json (hunk 21 above)
Comment (Prithvi-MSFT) [reply samples/app-link-unfurling-in-share-to-teams/csharp/assets/sample.json]: changed

**Review by ChetanSharma-msft — APPROVED** [review]
Changes looks fine, approving it.

**Comment by ChetanSharma-msft** [comment]
@Prithvi-MSFT - Kindly run build pipeline and share the link

**Comment by Prithvi-MSFT** [comment]
Build pipeline status - All builds passed
https://github.com/OfficeDev/Microsoft-Teams-Samples/actions/runs/4279633096

**Review by Wajeed-msft — APPROVED** [review]
Please check if this is already available in public rings.

**Comment by Prithvi-MSFT** [comment]
Closing this PR. Raised here https://github.com/OfficeDev/Microsoft-Teams-Samples/pull/273

**Pull request closed without merging**